presto-common/src/main/java/com/facebook/presto/common/array/Arrays.java

```diff
@@ -11,58 +11,67 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.facebook.presto.array;
+package com.facebook.presto.common.array;
 
 import com.facebook.presto.common.block.ArrayAllocator;
 
-import static com.facebook.presto.array.Arrays.ExpansionFactor.SMALL;
-import static com.facebook.presto.array.Arrays.ExpansionOption.INITIALIZE;
-import static com.facebook.presto.array.Arrays.ExpansionOption.NONE;
-import static com.facebook.presto.array.Arrays.ExpansionOption.PRESERVE;
+import static com.facebook.presto.common.array.Arrays.ExpansionFactor.SMALL;
+import static com.facebook.presto.common.array.Arrays.ExpansionOption.INITIALIZE;
+import static com.facebook.presto.common.array.Arrays.ExpansionOption.NONE;
+import static com.facebook.presto.common.array.Arrays.ExpansionOption.PRESERVE;
 import static java.lang.Math.max;
 import static java.lang.Math.min;
 
 public class Arrays
 {
     private Arrays() {}
 
-    public static int[] ensureCapacity(int[] buffer, int capacity)
+    public static boolean[] ensureCapacity(boolean[] buffer, int capacity)
+    {
+        if (buffer == null || buffer.length < capacity) {
+            return new boolean[(int) (capacity * SMALL.expansionFactor)];
+        }
+
+        return buffer;
+    }
+
+    public static byte[] ensureCapacity(byte[] buffer, int capacity)
     {
         return ensureCapacity(buffer, capacity, SMALL, NONE);
```

Comment:
No. Java does not support generics for primitive types. You could try T[] but T has to be a class e.g. Integer instead of int. You would have to rely on auto(un)boxing when you read the real data for every value. Here using primitive types is essential to performance and memory footprint. Java class objects are normally at least dozens of bytes even though the effective part is just one byte. You can check the size of a Byte object to see how big it is.